src/containerapp/azext_containerapp/containerapp_resiliency_decorator.py

```diff
@@ -0,0 +1,476 @@
+                      TcpConnectionPool as TcpConnectionPoolModel,
+                      HttpConnectionPool as HttpConnectionPoolModel)
+
+logger = get_logger(__name__)
+
+
+class ContainerAppResiliencyDecorator(BaseResource):
+    def __init__(self, cmd: AzCliCommand, client: Any, raw_parameters: Dict, models: str):
+        super().__init__(cmd, client, raw_parameters, models)
+
+    def get_argument_container_app_name(self):
+        return self.get_param("container_app_name")
+
+    def get_argument_disable_warnings(self):
+        return self.get_param("disable_warnings")
+
+    def get_argument_tcp_retry_max_connect_attempts(self):
+        return self.get_param("tcp_retry_max_connect_attempts")
+
+    def get_argument_circuit_breaker_consecutive_errors(self):
+        return self.get_param("circuit_breaker_consecutive_errors")
+
+    def get_argument_circuit_breaker_interval(self):
+        return self.get_param("circuit_breaker_interval")
+
+    def get_argument_circuit_breaker_max_ejection(self):
+        return self.get_param("circuit_breaker_max_ejection")
+
+    def get_argument_tcp_connection_pool_max_connections(self):
+        return self.get_param("tcp_connection_pool_max_connections")
+
+    def get_argument_http_connection_pool_http1_max_pending_req(self):
+        return self.get_param("http_connection_pool_http1_max_pending_req")
+
+    def get_argument_http_connection_pool_http2_max_req(self):
+        return self.get_param("http_connection_pool_http2_max_req")
+
+    def get_argument_timeout_response_in_seconds(self):
+        return self.get_param("timeout_response_in_seconds")
+
+    def get_argument_timeout_connection_in_seconds(self):
+        return self.get_param("timeout_connection_in_seconds")
+
+    def get_argument_http_retry_max(self):
+        return self.get_param("http_retry_max")
+
+    def get_argument_http_retry_delay_in_milliseconds(self):
+        return self.get_param("http_retry_delay_in_milliseconds")
+
+    def get_argument_http_retry_interval_in_milliseconds(self):
+        return self.get_param("http_retry_interval_in_milliseconds")
+
+    def get_argument_http_retry_status_codes(self):
+        return self.get_param("http_retry_status_codes")
+
+    def get_argument_http_retry_errors(self):
+        return self.get_param("http_retry_errors")
+
+    def get_argument_yaml(self):
+        return self.get_param("yaml")
+
+    def get_argument_default(self):
+        return self.get_param("default")
+
+    def set_argument_name(self, name):
+        self.set_param("name", name)
+
+    def set_argument_container_app_name(self, container_app_name):
+        self.set_param("container_app_name", container_app_name)
+
+    def validate_positive_argument(self, argument_name):
+        argument_value = getattr(self, f"get_argument_{argument_name}")()
+        if argument_value is not None and argument_value < 1:
+            raise ValidationError(f"--{argument_name} must be greater than 0")
+
```

Comment:
the arugment name is in format _ but for cli layout it should be -, right?

Reply:
Good catch, I'll change it.